pkgs/applications/networking/hpmyroom/default.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv, lib, fetchurl, rpmextract, autoPatchelfHook, makeWrapper
+, libuuid, libXtst, libXfixes, glib, gst_all_1, alsaLib, freetype, fontconfig
+, libXext, libGL, libpng, libXScrnSaver, libxcb, xorg, libpulseaudio, libdrm
+}:
+stdenv.mkDerivation rec {
+  pname = "hpmyroom";
+  version = "11.1.0.0508";
+
+  src = fetchurl {
+    url = "https://www.myroom.hpe.com/downloadfiles/${pname}-${version}.x86_64.rpm";
+    sha256 = "1j7mzvf349yxb42m8syh73gpvil01hy1a2wrr0rdzb2ijfnkxyaa";
+  };
+
+  nativeBuildInputs = [
+    rpmextract autoPatchelfHook makeWrapper
+  ];
+
+  buildInputs = let deps = [
+    libuuid libXtst libXScrnSaver libXfixes alsaLib freetype fontconfig libXext
+    libGL libpng libxcb libpulseaudio libdrm
+    glib  # For libgobject
+    stdenv.cc.cc  # For libstdc++
+  ] ++ (with gst_all_1; [ gstreamer gst-plugins-base ])
+    ++ (with xorg; [ libX11 ]);
+  in map (dep: dep.lib or dep.out or dep) deps;
+
+  unpackPhase = ''
+    rpmextract $src
+  '';
+
+  prePatch = ''
+    substituteInPlace usr/share/applications/HP-myroom.desktop \
+      --replace /usr/bin/hpmyroom hpmyroom \
+      --replace Icon=/usr/share/hpmyroom/Resources/MyRoom.png Icon=$out/share/hpmyroom/Resources/MyRoom.png
+  '';
+
+  installPhase = ''
+    mv usr $out
+    wrapProgram $out/bin/hpmyroom \
+      --set QT_XKB_CONFIG_ROOT "${xorg.xkeyboardconfig}/share/X11/xkb"
+    ln -s ${libpng}/lib/libpng.so $out/lib/hpmyroom/libpng15.so.15
+  '';
+
+  meta = {
+    description = "Client for HPE's MyRoom web conferencing solution";
+    maintainers = with lib.maintainers; [ johnazoidberg ];
+    license = lib.licenses.unfree;
+    homepage = "https://myroom.hpe.com";
+    # TODO: A Darwin binary is available upstream
+    platforms = [ "x86_64-linux" ];
+    hydraPlatforms = [];
```

Comment:
I don't think this is needed?

Reply:
I'm not sure. It cannot be redistributed so hydra must not build it. Is that given with `meta.license = lib.licenses.unfree;`?

Reply:
Yes, it will refuse to evaluate it when unfree. So it's unneeded.